native/src/widget/pane_grid.rs

```diff
@@ -897,3 +935,41 @@ fn hovered_split<'a>(
         })
         .next()
 }
+
+/// The visible contents of the [`PaneGrid`]
+#[derive(Debug)]
+pub enum Contents<T> {
+    /// All panes are visible
+    All(Vec<(Pane, T)>),
+    /// A maximized pane is visible
+    Maximized(Pane, T),
+}
+
+impl<T> Contents<T> {
+    /// Returns an iterator over the values of the [`Contents`]
+    pub fn iter(&self) -> Box<dyn Iterator<Item = (Pane, &T)> + '_> {
+        match self {
+            Contents::All(contents) => Box::new(
+                contents.iter().map(|(pane, content)| (*pane, content)),
+            ),
+            Contents::Maximized(pane, content) => {
+                Box::new(std::iter::once((*pane, content)))
+            }
+        }
+    }
+
+    fn iter_mut(&mut self) -> Box<dyn Iterator<Item = (Pane, &mut T)> + '_> {
+        match self {
+            Contents::All(contents) => Box::new(
+                contents.iter_mut().map(|(pane, content)| (*pane, content)),
+            ),
+            Contents::Maximized(pane, content) => {
+                Box::new(std::iter::once((*pane, content)))
+            }
+        }
+    }
```

Comment:
Great! We avoid a lot of code changes with this API :) Iterators are pretty nifty.